tests/integration/relay_pingpong_test.go

```diff
@@ -252,7 +251,7 @@ func (p player) IBCPacketReceive(_ wasmvm.Checksum, _ wasmvmtypes.Env, msg wasmv
 	nextValue := p.incrementCounter(lastBallSentKey, store)
 	newHit := NewHit(p.actor, nextValue)
 	respHit := &wasmvmtypes.IBCMsg{SendPacket: &wasmvmtypes.SendPacketMsg{
-		ChannelID: packet.Src.ChannelID,
+		ChannelID: packet.Dest.ChannelID,
```

Comment:
Wait a minute. Was this wrong all along? Nice catch! Why did it even work in the first place?

Reply:
Ah, that makes sense. Our ibctesting module was probably using a counter per chain while the ibc-go one uses a global one.
Anyway, this change is the correct one, since the pingpong mock is receiving the packet here, so the channel id on its side is the destination channel of the packet.